src/Tribe/Widget/Manager.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * Widgets manager for Tribe plugins.
+ *
+ * @package Tribe\Widget
+ * @since   TBD
+ */
+
+namespace Tribe\Widget;
+
+/**
+ * Class Widget Manager.
+ *
+ * @since  TBD
+ *
+ * @package Tribe\Widget
+ */
+class Manager {
+
+	/**
+	 * Current widgets.
+	 *
+	 * @since TBD
+	 *
+	 * @var array $current_widget An array containing the current widgets being executed.
+	 */
+	public $current_widget = [];
```

Comment:
I would avoid a `public` property and go with a more structured (and overrideable and deprecate-able) getter and setter API.
Make the `$current_widget` property `protected` and implement the `get_current_widget` (or just `current_widget`) and `set_current_widget` methods.
If the setter is not used anywhere, then skip it.